### PR TITLE
[Fabric-Sync] Initial version of fabric-sync example

### PR DESCRIPTION
**Pull request by yufengwangca**
Currently, conducting fabric synchronization requires running two separate example apps, Fabric-Admin and Fabric-Bridge-app, on each side. When integrating into the actual product, these apps are expected to run on the same physical device. This poses a challenge for certain products, especially those based on RTOS systems that do not support multiple processes, making it difficult to properly isolate the two apps.

To address this issue, we plan to develop a new FabricSync example app that combines the functionality of both Fabric-Admin and Fabric-Bridge-app within a single application, using a single process with a shared event loop.

This is the initial version of the FabricSync example app which function as both a Commissioner and a Commissionee simultaneously. Both roles will operate within the same process and utilize a shared event loop, allowing the application to handle events for both roles within a unified framework. This design leverages Matter's event-driven architecture, where a single event loop manages various network, system, and protocol events, regardless of whether the device acts as a Commissioner or Commissionee.

You can commission another instance of fabric-sync app running on another machine using the following command

` controller commission-onnetwork 20202021 3840 192.168.86.246 5540`

### CODE REVIEW

**Review by yunhanw-google — APPROVED** [review]

File: examples/fabric-sync/bridge/BUILD.gn [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import("${chip_root}/src/lib/lib.gni")
+
+config("config") {
+  include_dirs = [ "include" ]
+}
+
+chip_data_model("fabric-bridge-zap") {
+  zap_file = "fabric-bridge.zap"
+  is_server = true
+  cflags = [ "-DDYNAMIC_ENDPOINT_COUNT=16" ]
```
Comment (jmartinez-silabs) [inline examples/fabric-sync/bridge/BUILD.gn:26]: does it need to be a precompile define?

**Review by jmartinez-silabs — APPROVED** [review]

File: examples/fabric-sync/bridge/BUILD.gn (hunk 1 above)
Comment (yufengwangca) [reply examples/fabric-sync/bridge/BUILD.gn:26]: Acked

File: examples/fabric-sync/bridge/include/CHIPProjectAppConfig.h [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+#define CHIP_DEVICE_CONFIG_EXTENDED_DISCOVERY_TIMEOUT_SECS (24 * 60 * 60)
+
+// Advertise TV device type in DNS-SD
+#define CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONABLE_DEVICE_TYPE 1
+
+#define CHIP_DEVICE_CONFIG_DEVICE_TYPE 35 // 0x0023 = 35 = Video Player
+
+// Include device name in discovery for casting use case
+#define CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONABLE_DEVICE_NAME 1
+#define CHIP_DEVICE_CONFIG_DEVICE_NAME "Test TV"
```
Comment (tehampson) [inline examples/fabric-sync/bridge/include/CHIPProjectAppConfig.h:63]: Why is this the device name?

File: examples/fabric-sync/bridge/include/CHIPProjectAppConfig.h [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+// Enable app platform
+#define CHIP_DEVICE_CONFIG_APP_PLATFORM_ENABLED 1
+
+// overrides CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT in CHIPProjectConfig
+#define CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT 16
+
+// For casting, we need to allow more ACL entries, and more complex entries
```
[the remark this answers is not in the document]
Comment (yufengwangca) [reply examples/fabric-sync/bridge/include/CHIPProjectAppConfig.h:71]: CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT is copied form 
./fabric-bridge-app/fabric-bridge-common/include/CHIPProjectAppConfig.h

casting stuff need to be removed

File: examples/fabric-sync/bridge/include/CHIPProjectAppConfig.h (hunk 2 above)
Comment (yufengwangca) [reply examples/fabric-sync/bridge/include/CHIPProjectAppConfig.h:63]: need to be removed

**Pull request merged**